packages/core/src/physics/PhysicsScene.ts

```diff
@@ -374,10 +376,210 @@ export class PhysicsScene {
         return false;
       }
     } else {
-      return this._nativePhysicsScene.raycast(ray, distance, onRaycast);
+      return this._nativePhysicsScene.raycast(ray, distance, preFilter);
     }
   }
 
+  /**
+   * Casts a box through the Scene and returns true if there is any hit.
+   * @param center - The center of the box
+   * @param halfExtents - Half the size of the box in each dimension
+   * @param direction - The direction to sweep along
+   * @param orientation - The rotation of the box, default is identity quaternion (axis-aligned)
+   * @param distance - The max distance to sweep, default is Number.MAX_VALUE
+   * @param layerMask - Layer mask that is used to selectively ignore Colliders when sweeping, default is Layer.Everything
+   * @param outHitResult - Optional HitResult object to store detailed hit information
+   * @returns Returns true if the box intersects with any collider, otherwise false
+   */
+  boxCast(
+    center: Vector3,
+    halfExtents: Vector3,
+    direction: Vector3,
+    orientation: Quaternion = PhysicsScene._identityQuaternion,
+    distance: number = Number.MAX_VALUE,
+    layerMask: Layer = Layer.Everything,
+    outHitResult?: HitResult
+  ): boolean {
+    const preFilter = this._createPreFilter(layerMask);
+
+    const result = this._nativePhysicsScene.boxCast(
+      center,
+      orientation,
+      halfExtents,
+      direction,
+      distance,
+      preFilter,
+      outHitResult ? this._createSweepCallback(outHitResult) : undefined
+    );
+
```

Comment:
- result -> isInset is better?
- if result is false, `this._createSweepCallback` will not called?